libs/hdf-converters/src/veracode-mapper.ts

```diff
@@ -0,0 +1,587 @@
+}
+
+function formatDesc(input: unknown): string {
+  const text = [];
+  if (_.has(input, 'desc.para')) {
+    if (`${_.get(input, 'desc.para.text')}` !== 'undefined') {
+      text.push(`${_.get(input, 'desc.para.text')}`);
+    } else {
+      const len = Number(`${_.get(input, 'desc.para.length')}`);
+      for (let index = 0; index < len; index++) {
+        const paratext = `desc.para[${index}].text`;
+        text.push(`${_.get(input, paratext)}`);
+      }
+    }
+  }
+  return text.join('\n');
+}
+
+function cweloop(input: Record<string, unknown>, index: integer): string {
+  const categories = [
+    'pcrirelated',
+    'owasp',
+    'owasp2013',
+    'sans',
+    'certc',
+    'certccp',
+    'certjava',
+    'owaspmobile'
+  ];
+  let cwe = 'CWE-' + `${_.get(input, `cwe[${index}].cweid`)}` + ': ';
+  const cwename = `cwe[${index}].cwename`;
+  cwe += `${_.get(input, cwename)}`;
+  categories.forEach(function (value) {
+    if (_.has(input, `cwe[${index}].${value}`)) {
+      cwe += `; ${value}: ` + `${_.get(input, `cwe[${index}].${value}`)}`;
+    }
+  });
+  return cwe;
+}
+
+function formatCweData(input: Record<string, unknown>): string {
+  const text = [];
+  if (_.has(input, 'cwe')) {
+    const len = Number(`${_.get(input, CWE_LENGTH)}`);
+    for (let index = 0; index < len; index++) {
+      const cwe = cweloop(input, index);
+      text.push(cwe);
+    }
+  }
+  return text.join('\n');
+}
+
+function formatCweDesc(input: Record<string, unknown>): string {
+  const text = [];
+  if (_.has(input, 'cwe')) {
+    const len = Number(`${_.get(input, CWE_LENGTH)}`);
+    for (let index = 0; index < len; index++) {
+      const cweid = `cwe[${index}].cweid`;
+      let cwe = 'CWE-'.concat(`${_.get(input, cweid)}`) + ': ';
+      const cwename = `cwe[${index}].cwename`;
+      const desc = `cwe[${index}].description.text.text`;
+      cwe += `${_.get(input, cwename)}` + ': ';
+      cwe += `${_.get(input, desc)}`;
+      text.push(cwe);
+    }
+  }
+  return text.join('\n');
+}
+
+function formatSourceLocation(input: Record<string, unknown>): string {
+  const text = [];
+  const len = Number(`${_.get(input, 'length')}`);
+  for (let index = 0; index < len; index++) {
+    const source = `[${index}].sourcefile`;
+    text.push(`${_.get(input, source)}`);
+  }
+  return text.join('\n');
+}
+
+function stringifyCodeDesc(input: Record<string, unknown>): string {
+  const categories = [
+    ['Line Number', 'line'],
+    ['Affect Policy Compliance', 'affects_policy_compliance'],
+    ['Remediation Effort', 'remediationeffort'],
+    ['Exploit level', 'exploitLevel'],
+    ['Issue ID', 'issueid'],
+    ['Module', 'module'],
+    ['Type', 'type'],
+    ['CWE ID', 'cweid'],
+    ['Date First Occurence', 'date_first_occurence'],
+    ['CIA Impact', 'cia_impact'],
+    ['Description', 'description']
+  ];
+  let flawDesc =
+    'Sourcefile Path: ' + `${_.get(input, 'sourcefilepath')}` + ';';
+  categories.forEach(function (value) {
```

Comment:
i recommend `for of` loops over foreach when es6 is available (which it is)

https://2ality.com/2021/01/looping-over-arrays.html - really solid article explaining the different for loop structures in js

anyways, i'd still recommend map/reduce (with the reduce being the built-in join func) here